libmscore/staff.h

```diff
@@ -185,6 +186,8 @@ class Staff final : public ScoreElement {
       QMap<int,int>* channelList(int voice) { return  &_channelList[voice]; }
       SwingParameters swing(int tick)  const;
       QMap<int,SwingParameters>* swingList() { return &_swingList; }
+      QMap<int,int>* capoList() { return &_capoList; }
```

Comment:
Why not to make `_capoList` public variable? This method gives direct access to the private data member which breaks incapsulation. If you use this member to only insert or clear elements, implement these methods in the `Staff` class like:
`insertCapo` and `clearCapo`.

Reply:
@anatoly-os In that case, surely the `swingList` method also breaks encapsulation? It returns the private `_swingList`.

Reply:
It is the boyscouts principle: "make the code a bit better after your changes". And surely don't make it worse.
Getting access to private data members is bad coding practice. It is better to make data members public.
Anyway, new code entries should improve the code and make it better. IMHO

Reply:
Well, that's true, they should. But this is a common thing across the entire MuseScore codebase, I think? Also, is it just the swingList and capoList methods you're talking about, or is every function that returns the value of a private variable (rather than a pointer to one) breaking encapsulation as well?

Reply:
It is not me personally, but also Meyers and Stroustrup and others. Returning non constant references or pointers to private data members allows changing the state of the class externally. Returning const references or pointers or copies of the objects don't allow changing class state externally, so it is safe.

Regarding codebase, this approach allows writing more code in less time and makes sense if you are only one contributor. Things are changing, so contributing more or less safe and well designed code is a priority.

Reply:
Making the method const won't allow you to to call non const insert/clear. You need to remove these methods and implement wrappers instead like this:
`
void insertCapo(int tick, Capo* capo) {
      _capoList.insert(tick, capo);
      }
`
I'm not sure about the `Capo*` pointer type, I'm from mobile and cannot quickly find the type of the `st->capo()` returning value.

Reply:
OK, I'm on mobile too so I will do this when I can.